spectrometer.py

```diff
@@ -1,303 +1,331 @@
+
+
+def inform_user_of_exposure(max_result):
+    exposure = max_result / (255 + 255 + 255)
+    print("ideal exposure between 0.15 and 0.30")
+    print("exposure=", exposure)
+    if exposure < 0.15:
+        print("consider increasing shutter time")
+    elif exposure > 0.3:
+        print("consider reducing shutter time")
+
+
+def save_image_with_overlay(im, name):
+    output_filename = name + "_out.jpg"
+    ImageFile.MAXBLOCK = 2 ** 20
+    im.save(output_filename, "JPEG", quality=80, optimize=True, progressive=True)
+
+
+def normalize_results(results, max_result):
+    for wavelength in results:
+        results[wavelength] = results[wavelength] / max_result
+    return results
+
+
+def export_csv(name, normalized_results):
+    csv_filename = name + ".csv"
+    csv = open(csv_filename, 'w')
+    csv.write("wavelength,amplitude\n")
+    for wavelength in normalized_results:
+        csv.write(wavelength)
+        csv.write(",")
+        csv.write("{:0.3f}".format(normalized_results[wavelength]))
+        csv.write("\n")
+    csv.close()
+
+
+def export_diagram(name, normalized_results):
+    antialias = 4
+    w = 600 * antialias
+    h2 = 300 * antialias
+
+    h = h2 - 20 * antialias
+    sd = Image.new('RGB', (w, h2), (255, 255, 255))
+    draw = ImageDraw.Draw(sd)
+
+    w1 = 380.0
+    w2 = 780.0
+    f1 = 1.0 / w1
+    f2 = 1.0 / w2
+    for x in range(0, w, 1):
+        # Iterate across frequencies, not wavelengths
+        lambda2 = 1.0 / (f1 - (float(x) / float(w) * (f1 - f2)))
+        c = wavelength_to_color(lambda2)
+        draw.line((x, 0, x, h), fill=c)
+
+    pl = [(w, 0), (w, h)]
+    for wavelength in normalized_results:
+        wl = float(wavelength)
+        x = int((wl - w1) / (w2 - w1) * w)
+        # print wavelength,x
+        pl.append((int(x), int((1 - normalized_results[wavelength]) * h)))
+    pl.append((0, h))
+    pl.append((0, 0))
+    draw.polygon(pl, fill="#FFF")
+    draw.polygon(pl)
+
+    font = ImageFont.truetype('/usr/share/fonts/truetype/lato/Lato-Regular.ttf', 12 * antialias)
+    draw.line((0, h, w, h), fill="#000", width=antialias)
+
+    for wl in range(400, 1001, 10):
+        x = int((float(wl) - w1) / (w2 - w1) * w)
+        draw.line((x, h, x, h + 3 * antialias), fill="#000", width=antialias)
+
+    for wl in range(400, 1001, 50):
+        x = int((float(wl) - w1) / (w2 - w1) * w)
+        draw.line((x, h, x, h + 5 * antialias), fill="#000", width=antialias)
+        wls = str(wl)
+        tx = draw.textsize(wls, font=font)
+        draw.text((x - tx[0] / 2, h + 5 * antialias), wls, font=font, fill="#000")
+
+    # save chart
+    sd = sd.resize((int(w / antialias), int(h / antialias)), Image.ANTIALIAS)
+    output_filename = name + "_chart.png"
+    sd.save(output_filename, "PNG", quality=95, optimize=True, progressive=True)
+
+
+def main():
+    # 1. Take picture
+    camera = picamera.PiCamera()
+    name = sys.argv[1]
```

Comment:
I recommend validating that sys.argv actually has 2 arguments here and print a helpful usage message if not. (You could use argparse to do help with this.)